File: FriendsCoolWater/Models/TeamModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FriendsCoolWater.Models
{
    [Table("Teams")]
    public class TeamModel
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; }

        [MaxLength(100)]
        public string Description { get; set; }

        [Required]
        public bool Active { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime CreatedOn { get; set; }

        [Required]
        [MaxLength(50)]
        public string CreatedBy { get; set; }
        
        [DataType(DataType.DateTime)]
        public DateTime? ModifiedOn { get; set; }
        
        [MaxLength(50)]
        public string ModifiedBy { get; set; }

        public virtual ICollection<TeamEmployeeModel> TeamEmployees { get; set; }
    }
}

File: FriendsCoolWater/ClientApp/src/app/core/team/team.model.ts
export interface Team {
  id?: number;
  name: string;
  description: string;
  active: boolean;
  createdOn: Date;
  createdBy: string;
  modifiedOn?: Date;
  modifiedBy: string;
}

export interface EmployeesInTeams {
  id: number;
  teamId: number,
  employeeId: string,
  teamName: string,
  employeeName: string  
}
